src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/RuntimeExports.cs

```diff
@@ -392,26 +392,22 @@ internal static unsafe IntPtr RhGetRuntimeHelperForType(MethodTable* pEEType, Ru
                         return (IntPtr)(delegate*<MethodTable*, object>)&InternalCalls.RhpNewFast;
 
                 case RuntimeHelperKind.IsInst:
-                    if (pEEType->IsArray)
-                        return (IntPtr)(delegate*<MethodTable*, object, object>)&TypeCast.IsInstanceOfArray;
-                    else if (pEEType->HasGenericVariance)
-                        return (IntPtr)(delegate*<MethodTable*, object, object>)&TypeCast.IsInstanceOf;
+                    if (pEEType->IsArray || pEEType->HasGenericVariance)
```

Comment:
We should not need the `IsArray` check here. It is covered by the `IsParameterizedType` check below.

This can be even possibly changed to `if (pEEType->HasGenericVariance || pEEType->IsParameterizedType || pEEType->IsFunctionPointerType)` so that there is only a one path that returns the Any helper.